staging_service/AutoDetectUtils.py

```diff
@@ -2,39 +2,53 @@
 This class is in charge of determining possible importers by determining the suffix of the filepath pulled in,
 and by looking up the appropriate mappings in the supported_apps_w_extensions.json file
 """
-from typing import Optional
+from typing import Optional, Tuple
 
 
 class AutoDetectUtils:
     _FILE_EXTENSION_MAPPINGS = None  # expects to be set by config
     _MAPPINGS = None  # expects to be set by config
 
     @staticmethod
-    def determine_possible_importers(filename: str) -> Optional[list]:
+    def determine_possible_importers(filename: str) -> Tuple[Optional[list], str, Optional[str]]:
         """
         Given a filename, come up with a reference to all possible apps.
         :param filename: The filename to find applicable apps for
-        :return: A list of mapping references, or None if not found
+        :return: A tuple containing:
+            a list of mapping references, or None if not found
+            the file prefix
+            the file suffix, if a suffix matched a mapping
         """
         dotcount = filename.count(".")
         if dotcount:
             # preferentially choose the most specific suffix (e.g. longest)
             # to get file type mappings
             for i in range(1, dotcount + 1):
-                suffix = filename.split(".", i)[-1].lower()
+                parts = filename.split(".", i)
+                suffix = parts[-1].lower()
                 if suffix in AutoDetectUtils._MAPPINGS["types"]:
-                    return AutoDetectUtils._MAPPINGS["types"][suffix]
-        return None
+                    prefix = ".".join(parts[0:i])
+                    return AutoDetectUtils._MAPPINGS["types"][suffix], prefix, parts[-1]
+        return None, filename, None
 
     @staticmethod
     def get_mappings(file_list: list) -> dict:
         """
         Given a list of files, get their mappings if they exist
         :param file_list: A list of files
-        :return: return a listing of apps and a listing of extension_mappings for each filename
+        :return: return a listing of apps, a listing of extension_mappings for each filename,
+            and information about each file, currently the file prefix and the suffix used to
+            determine the mappings
         """
         mappings = []
+        fileinfo = []
         for filename in file_list:
-            mappings.append(AutoDetectUtils.determine_possible_importers(filename))
-        rv = {"apps": AutoDetectUtils._MAPPINGS["apps"], "mappings": mappings}
+            typemaps, prefix, suffix = AutoDetectUtils.determine_possible_importers(filename)
+            mappings.append(typemaps)
+            fileinfo.append({"prefix": prefix, "suffix": suffix})
+        rv = {
+            "apps": AutoDetectUtils._MAPPINGS["apps"],
```

Comment:
I don't think the "apps" are being used by any FE code. We could just disable that return.

Reply:
that'd be great - removing all that code would really simplify the mappings code. I've pinged the FE devs - if they agree I'll do it in another PR.

Reply:
How would removing the return simplify the mappings code? Where should the source of truth for the mapping be, in the narrative config file or in the staging service? When we added the mappings here, we didn't realize they were already maintaining a list in the narrative.

Reply:
> How would removing the return simplify the mappings code?

Because we could delete some or all of this:
https://github.com/kbase/staging_service/blob/develop/staging_service/autodetect/GenerateMappings.py#L34-L117
https://github.com/kbase/staging_service/blob/develop/staging_service/autodetect/GenerateMappings.py#L161-L186

> Where should the source of truth for the mapping be, in the narrative config file or in the staging service?

I don't really have a strong feeling either way, but having it in the narrative seems a reasonable a place as any. The narrative does all the front end app handling other than this mapping (which it doesn't use anyway) so  it seems sensible to keep the mapping config there as well, and have the staging service just deal with file types / categories.